gitlabform/gitlabform/processors/project/environments_processor.py

```diff
@@ -0,0 +1,64 @@
+import logging
+import json
+import operator
+
+from gitlabform.gitlab import GitLab
+from gitlabform.gitlabform.processors.abstract_processor import AbstractProcessor
+
+
+class EnvironmentsProcessor(AbstractProcessor):
+    def __init__(self, gitlab: GitLab):
+        super().__init__("environments")
+        self.gitlab = gitlab
+
+    def _process_configuration(self, project_and_group: str, configuration: dict):
+        logging.debug(
+            "Environments BEFORE: %s",
+            self.gitlab.get_all_environments(project_and_group),
+        )
+        e = self.gitlab.get_all_environments(project_and_group)
+        for environment in sorted(configuration["environments"]):
+            logging.info("Setting environment: %s", environment)
+            data = configuration["environments"][environment]
+            env = data["name"]
+
+            if env in map(operator.itemgetter("name"), e):
```

Comment:
I like that you are getting all the envs and then operate only the ones that you should - that minimizes the number of calls to GitLab. 🙂 It also opens room to easily add "enforce" option - that would keep the envs exactly as defined in the code, so deleting all the ones that may have been defined manually - in the future. 👍